packages/medusa/src/api-v2/utils/middlewares/pricing/set-pricing-context.ts

```diff
@@ -0,0 +1,88 @@
+import { MedusaPricingContext } from "@medusajs/types"
+import { isPresent, MedusaError } from "@medusajs/utils"
+import { NextFunction } from "express"
+import { MedusaRequest } from "../../../../types/routing"
+import { refetchEntities, refetchEntity } from "../../refetch-entity"
+
+export function setPricingContext() {
+  return async (req: MedusaRequest, _, next: NextFunction) => {
+    // If the endpoint doesn't request prices, we can exit early
+    if (
+      !req.remoteQueryConfig.fields.some((field) =>
+        field.startsWith("variants.prices")
```

Comment:
This check won't work if we fetch a variant directly, for example. Do we even need the check? Should we leave it to the one applying the middleware to decide on the check (eg. by passing a func)?

Reply:
Yup, this is specific to products, will change the name of the middleware